tests/drivers/gpio/gpio_nrf/testcase.yaml

```
@@ -0,0 +1,9 @@
+common:
+  tags: drivers gpio
+  depends_on: gpio
+  harness: ztest
+tests:
+  drivers.gpio.gpio_nrf:
+    platform_allow:
+      - nrf54l15pdk/nrf54l15/cpuapp
+      - nrf54h20dk/nrf54h20/cpuapp
```

It should be possible to execute this test on any nRF platform, so I'd suggest to use:
```suggestion
    filter: dt_compat_enabled("nordic,nrf-gpio")
```

---

I need to revert  this due to:
110/445 raytac_mdbt53v_db_40/nrf5340/cpunet tests/drivers/gpio/gpio_nrf/drivers.gpio.gpio_nrf   ERROR Build failure (build)

Not all NRF DKs have 'led0' alias definedm this is the reason of this error.